### PR TITLE
Merging the ManagedNetworkFabric changes in RPSaaSMaster into main

### PR DESCRIPTION
**Pull request by Mohana-Krishna-Nali**
The below given PRs are approved and merged into the RPSaaSMaster. 

1. https://github.com/Azure/azure-rest-api-specs-pr/pull/11537
2. https://github.com/Azure/azure-rest-api-specs-pr/pull/11749
3. https://github.com/Azure/azure-rest-api-specs-pr/pull/12117

This PR contains the combination of the all the above three PRs.

### CODE REVIEW

**Review by sylar217 — APPROVED** [review]

**Comment by ms-zhenhua** [comment]
@Mohana-Krishna-Nali  Please reach out to Azure Breaking Changes Reviewers azbreakchangereview@microsoft.com to get a breaking changes approval and ARM can review the changes afterwards. Once you have the approval, please remove the "ARMChangesRequested" label from the PR to make the PR visible to the ARM reviewers.

**Comment by mikekistler** [comment]
Breaking changes previously reviewed and approved.

**Pull request merged**
